backend/items/models.py

```
@@ -376,3 +409,149 @@ def donated_badge_progress(user):
+    class Meta:
+        unique_together = [('user', 'slug')]
+        indexes = [
+            models.Index(fields=['user', 'full_path']),
+            models.Index(fields=['full_path']),
+        ]
+        ordering = ['full_path']
+
+    def __str__(self):
+        return f"{self.full_path} ({self.user.username})"
+
+    def save(self, *args, **kwargs):
```

Ruff (0.15.2)

[warning] Missing type annotation for `*args`

(ANN002)